filebeat/prospector/prospectors/registry.go

```diff
@@ -0,0 +1,45 @@
+package prospectors
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input/file"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type ProspectorFactoryData struct {
```

Comment:
how about naming it 'Context' or 'Environment'? 

Keep in mind, the package names is part of a type its full. Try to reduce stuttering if possible. e.g.

`prospectors.ProspectorFactory` can be minimized to `prospectors.Factory`. And `prospectors.ProspectorFactoryData` to `prospectors.Context`.